### PR TITLE
Support for more LTI-options

### PR DESCRIPTION
**Pull request by alalav1**
Added options 'lti_aplus_get_and_post' and 'lti_open_in_iframe' . This branch also has status updates from   #36 included. Continued the previous pull request #31 authored by @piehei

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -371,7 +373,10 @@ It accepts the following options:
   the LTI service that must be configured in the A+ site beforehand.
 * `lti_resource_link_id`: LTI exercise key
 * `lti_context_id`: LTI course key
-
+* `lti_open_in_iframe`: Open the exercise in an iframe inside the A+ page instead of a new window.
+  Doesn't take any parameters.
+* `lti_aplus_get_and_post`: Perform GET and POST from A+ to custom service URL with LTI data appended.
```
[the remark this answers is not in the document]
Comment (Mankro) [reply README.md]: You could check out how the aplus-manual covers these options and also update the manual if it is outdated.

File: README.md [hunk 2]
```diff
@@ -389,6 +394,9 @@ The `url` option may exclude the domain of the service URL since the domain
 must be equal to the URL defined in the LTI service anyway.
 The LTI parameters may be defined in the config.yaml file linked with
 the `config` option, thus it is not required to define them in RST.
+However if the config.yaml is defined in the submit directive, all lti-options
+in the submit directive are ignored and they must be written in the config.yaml
+instead.
```
Comment (Mankro) [inline README.md]: I think that this section must be made more clear about the two methods for defining LTI exercises: `submit` directive with the config option and using LTI options in the YAML file OR using LTI options in the submit directive with no config option. For example, using a list could make the alternatives more clear to the reader and differentiate them properly.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alalav1) [reply README.md]: Ok. That was the description in A-plus which I just copied directly.

File: README.md [hunk 3]
```diff
@@ -371,6 +373,11 @@ It accepts the following options:
   the LTI service that must be configured in the A+ site beforehand.
 * `lti_resource_link_id`: LTI exercise key
 * `lti_context_id`: LTI course key
+* `lti_open_in_iframe`: Open the exercise in an iframe inside the A+ page instead of a new window.
+  Doesn't take any parameters.
+* `lti_aplus_get_and_post`:  The exercise uses the A+ protocol to connect to the service.
```
Comment (Mankro) [inline README.md]: get and post line has now two spaces after `:`

File: README.md [hunk 4]
```diff
@@ -371,6 +373,11 @@ It accepts the following options:
...
+  The LTI launch parameters are appended to the A+ protocol parameters. This does not work with standard LTI services.
+  Doesn't take any parameters.
```
Comment (Mankro) [inline README.md]: Convert "Doesn't take any parameters."  into "This option does not take any parameters."

File: README.md [hunk 5]
```diff
@@ -387,8 +394,12 @@ The LTI service must be configured beforehand in A+ by an administrator.
 The `lti` option refers to the label of the LTI service.
 The `url` option may exclude the domain of the service URL since the domain
 must be equal to the URL defined in the LTI service anyway.
-The LTI parameters may be defined in the config.yaml file linked with
-the `config` option, thus it is not required to define them in RST.
+There is two ways to define LTI exercise:
```
Comment (Mankro) [inline README.md]: grammar: There are two ways.

File: README.md [hunk 6]
```diff
@@ -387,8 +394,12 @@ The LTI service must be configured beforehand in A+ by an administrator.
...
+
+* Use config.yaml file linked with `config` option and define LTI options there.
```
Comment (Mankro) [inline README.md]: using **the**: Use config.yaml file linked with **the** `config` option and define LTI options there. ... in **the** submit directive

File: README.md [hunk 7]
```diff
@@ -387,8 +394,12 @@ The LTI service must be configured beforehand in A+ by an administrator.
...
+  In this case, all LTI options written in submit directive will be ignored.
+* Write LTI options in submit directive. Remark that in this case submit directive must
+  not have `config` option set.
```
Comment (Mankro) [inline README.md]: **the** submit directive must not have **the** `config` option set

File: README.md [hunk 8]
```diff
@@ -387,8 +394,12 @@ The LTI service must be configured beforehand in A+ by an administrator.
 The `lti` option refers to the label of the LTI service.
 The `url` option may exclude the domain of the service URL since the domain
 must be equal to the URL defined in the LTI service anyway.
-The LTI parameters may be defined in the config.yaml file linked with
-the `config` option, thus it is not required to define them in RST.
+There are two ways to define LTI exercise:
+
+* Using the config.yaml file linked with the `config` option and defining LTI options there.
+  In this case, all LTI options written in the submit directive will be ignored.
+* Writing LTI options in submit directive. Remark that in this case the submit directive must
```
Comment (Mankro) [inline README.md]: One more time :) "in **the** submit directive"

**Comment by Mankro** [comment]
This looks good to me now. The PR #36 must be merged first since its commit is hanging here as well. Then, the commit history needs to be cleaned (rebase and squash).

In the final squashed commit, we should also give credit to Pietari as co-author.

**Comment by raphendyr** [comment]
Rebase and add credit to Pietari, as mentioned.

For reference: https://help.github.com/en/articles/creating-a-commit-with-multiple-authors

**Pull request merged**